app/views/articles/show.html.erb

```diff
@@ -11,29 +11,29 @@
     
     <div class="article-show-vote" >
       <% if current_user %>
-        <% if current_user.votes.where(article: @article).any? %>
-          <%= up_vote_links(@article.id, :put) %>
-          <div id="vote-count-<%= @article.id %>" class="article-vote-count"><%= @article.votes.sum(:value) %></div>
-          <%= down_vote_links(@article.id, :put) %>
+        <% if @article.votes.where(user: current_user).any? %>
+          <%= up_vote_links("Article", @article.id, :put) %>
+          <div id="vote-Article-count-<%= @article.id %>" class="article-vote-count"><%= @article.votes.sum(:value) %></div>
+          <%= down_vote_links("Article", @article.id, :put) %>
         <% else %>
-          <%= up_vote_links(@article.id, :post) %>
-          <div id="vote-count-<%= @article.id %>" class="article-vote-count"><%= @article.votes.sum(:value) %></div>
-          <%= down_vote_links(@article.id, :post) %>
+          <%= up_vote_links("Article", @article.id, :post) %>
+          <div id="vote-Article-count-<%= @article.id %>" class="article-vote-count"><%= @article.votes.sum(:value) %></div>
+          <%= down_vote_links("Article", @article.id, :post) %>
         <% end %>
       <% else %>
-        <%= link_to("", nil , class: "article-vote-top") %>
+        <%= link_to("", nil, id: "vote-button-up-nil", class: "article-vote-top") %>
         <div id="vote-count-<%= @article.id %>" class="article-vote-count"><%= @article.votes.sum(:value)  %></div>
-        <%= link_to("", nil , class: "article-vote-bottom") %> 
+        <%= link_to("", nil , id: "vote-button-down-nil", class: "article-vote-bottom") %> 
```

Comment:
I would wager that the ID of `vote-button-down-nil` is not unique in the DOM, and therefore should not a valid ID. What do you think of making some of this stuff a class?

Reply:
Changed it to include the voteable type